src/generic/clipboard/paste-component/index.jsx

```diff
@@ -0,0 +1,66 @@
+import { useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import { OverlayTrigger, Popover } from '@openedx/paragon';
+
+import { PopoverContent, PasteButton, WhatsInClipboard } from './components';
+import { clipboardPropsTypes, OVERLAY_TRIGGERS } from './constants';
+
+const PasteComponent = ({
+  onClick, clipboardData, text, className,
+}) => {
+  const [showPopover, togglePopover] = useState(false);
+  const popoverElementRef = useRef(null);
+
+  const handlePopoverToggle = (isOpen) => togglePopover(isOpen);
+
+  const renderPopover = (props) => (
+    <div role="link" ref={popoverElementRef} tabIndex="0">
+      <Popover
+        className="clipboard-popover"
+        id="popover-positioned"
+        onMouseEnter={() => handlePopoverToggle(true)}
+        onMouseLeave={() => handlePopoverToggle(false)}
+        onFocus={() => handlePopoverToggle(true)}
+        onBlur={() => handlePopoverToggle(false)}
+        {...props}
+      >
+        {clipboardData && (
+          <PopoverContent clipboardData={clipboardData} />
+        )}
+      </Popover>
+    </div>
+  );
+
+  return (
+    <>
+      <PasteButton className={className} onClick={onClick} text={text} />
+      <OverlayTrigger
+        show={showPopover}
+        trigger={OVERLAY_TRIGGERS}
```

Comment:
If we have `<OverlayTrigger trigger={OVERLAY_TRIGGERS}` and `OVERLAY_TRIGGERS = ['hover', 'focus']` then why do `<WhatsInClipboard>` and the `<Popover>`  above both have their own `onMouseEnter`, `onMouseLeave`, `onFocus`, `onBlur` handlers? There seems to be a lot of custom code here just to display a tooltip, and it seems a bit buggy in my testing. Can we rely more on Paragon and use less custom code here? I feel like there should already be a reusable interactive tooltip widget that doesn't require writing your own hover/focus handlers.

For reference, the legacy implementation doesn't use any JavaScript at all - just the `:hover` and `:focus-within` styles to make it appear.

Reply:
I agree that it looks with excessive logic and processing, but this was done to keep the tooltip in an open state and the ability to click on it, since the entire body of the tooltip is a link to the unit from which the component was copied

Reply:
I agree with @ihor-romaniuk. A generic tooltip is not supposed to be used when there is a clickable link inside. The alternative to a tooltip is a `ModalPopover`, but it was decided not to use that because of the z-indexing and because it requires the user to click when we want the open state to be available on hover.